docs/bigquery/snippets.py

```diff
@@ -90,6 +91,40 @@ def delete(self):
         self._wrapped.close()
 
 
+def test_create_client_default_credentials():
+    """Create a BigQuery client with Application Default Credentials"""
+
+    # [START bigquery_client_default_credentials]
+    from google.cloud import bigquery
+
+    # If you don't specify credentials when constructing the client, the
+    # client library will look for credentials in the environment.
+    client = bigquery.Client()
+    # [END bigquery_client_default_credentials]
+
+    assert client is not None
+
+
+def test_create_client_json_credentials():
+    """Create a BigQuery client with Application Default Credentials"""
+    with open(os.environ['GOOGLE_APPLICATION_CREDENTIALS']) as creds_file:
+        creds_file_data = creds_file.read()
+
+    open_mock = mock.mock_open(read_data=creds_file_data)
+
+    with mock.patch('io.open', open_mock):
```

Comment:
Fancy! I like this io mocking magic.

Do these snippets tests pass on Python 2.7? I know some I/O methods got moved around for Python 3.

Reply:
Ripped directly from python-docs-samples. Just checked and it's passing snippets tests for Python 2.7.